### PR TITLE
Create a map template for tests and use an empty space level for placing it.

### PR DESCRIPTION
**Pull request by warriorstar-orion**
## What Does This PR Do
This PR creates a dedicated map template for tests, and uses the empty space z-level as the level to place it on in order to perform tests.
## Why It's Good For The Game
Having an actual area with actual floors, that isn't a specialized admin area, or space, is more appropriate for 99% of our tests so far. I'm sure some will involve testing things in space but I imagine that to be the exception rather than the rule. This also means that I don't need to futz with internal area variables for e.g. placing mountable wallbumps in areas that don't require power, or are in a space area, and thus invalid.
## Images of changes

The landmarks set the bottom left and top right boundaries of the place for tests to create atoms. There is a one-tile margin between them and the walls of the map for sanity's sake, as many tests will end up in a random position and want to interact with nearby tiles in any direction.

2024_12_23__13_11_43__paradise dme  game_tests dmm  - StrongDMM

## Testing
Refactored tests to use new area, ran tests locally.

<hr>

### Declaration

- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog
NPFC

### CODE REVIEW

File: code/tests/atmos/test_ventcrawl.dm [hunk 1]
```diff
@@ -7,14 +7,16 @@
 /datum/milla_safe/ventcrawl_test_setup
 
 /datum/milla_safe/ventcrawl_test_setup/on_run(datum/game_test/ventcrawl/test)
+	// I'm sure get_area_turfs is totally deterministic and this will never go wrong
+	var/turf/run_loc_bottom_left = test.available_turfs[1]
```
Comment (lewcc) [inline code/tests/atmos/test_ventcrawl.dm:11]: Why not add some landmarks? Might help make things more deterministic

**Comment by Kenionatus** [comment]
Is the centcom level really a good testing area? It has teleport protection and possibly other things that could screw with tests. Wouldn't it make more sense to load it in at runtime on a normal Z level similar to how ruins are?

File: code/tests/atmos/test_ventcrawl.dm (hunk 1 above)
Comment (warriorstar-orion) [reply code/tests/atmos/test_ventcrawl.dm:11]: Done, PTAL

File: code/tests/_game_test.dm [hunk 2]
```diff
@@ -47,23 +47,26 @@
...
 	var/list/fail_reasons
 
 /datum/game_test/New()
-	run_loc_bottom_left = locate(1, 1, 1)
-	run_loc_top_right = locate(5, 5, 1)
+	available_turfs = list()
+	for(var/turf/T in get_area_turfs(/area/game_test))
+		if(istype(T, /turf/simulated/wall/indestructible/riveted))
+			continue
+		available_turfs |= T
```
[the remark this answers is not in the document]
Comment (warriorstar-orion) [reply code/tests/_game_test.dm]: Fixed, PTAL

**Comment by warriorstar-orion** [comment]
> Is the centcom level really a good testing area? It has teleport protection and possibly other things that could screw with tests. Wouldn't it make more sense to load it in at runtime on a normal Z level similar to how ruins are?

@Kenionatus  The testing map is now loaded on an available empty space level, PTAL

File: _maps/map_files/generic/centcomm.dmm [hunk 3]
```diff
@@ -424,9 +424,7 @@
 /turf/simulated/floor/plasteel/freezer,
 /area/ninja/holding)
 "bO" = (
-/obj/structure/closet/secure_closet/bar{
-	req_access = list(25)
-	},
+/obj/structure/closet/secure_closet/bar,
```
Comment (1080pCat) [inline _maps/map_files/generic/centcomm.dmm:427]: Whats going on here? Seems unrelated/unintentional?
Messing with this cabinet for reference
image

removing bar access for double reference.
image

Comment (warriorstar-orion) [reply _maps/map_files/generic/centcomm.dmm:427]: SDMM sanitizing variables

**Review by DGamerL — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]
